hw/ip/otbn/data/base-insns.yml

```diff
@@ -443,18 +443,22 @@
       doc: Number of instructions in the loop body
   straight-line: false
   doc: |
-    Repeats a sequence of code multiple times. The number of iterations is
-    read from `grs`, treated as an unsigned value. The number of
-    instructions in the loop is given in the `bodysize` immediate.
+    Repeats a sequence of code multiple times.
+    The number of iterations is read from `grs`, treated as an unsigned value.
+    The number of instructions in the loop is given in the `bodysize` immediate.
 
     The `LOOP` instruction doesn't support a zero iteration count.
     If the value in `grs` is zero, OTBN stops with the `ErrCodeLoop` error.
-    Starting a loop pushes an entry on to the loop stack.
+    Starting a loop pushes an entry on to the [loop stack](../#loop-stack).
     If the stack is already full, OTBN stops with the `ErrCodeLoop` error.
 
     There are also conditions on the code inside a loop.
-    A `LOOP`, `LOOPI`, jump or branch instruction may not appear as the last instruction in a loop.
+    `LOOP`, `LOOPI`, jump and branch instructions are all permitted inside a loop but may not appear as the last instruction in a loop.
     OTBN will stop on that instruction with the `ErrCodeLoop` error.
+
+    It is the programmer's responsibility to ensure loops are [well nested](../#loop-nesting).
+    Nested loops cannot end on the same instruction.
+    No errors are reported on badly nested loops.
```

Comment:
I'm not keen on this addition, since it's no longer describing the behaviour of the instruction but is rather giving some general advice to programmers.

You're already adding this text to the loop nesting section of `_index.md`; I don't really see the need to duplicate it.

Reply:
The `cannot end on the same instruction` is duplicated (accidentally in fact, wrote that before I decided to add the loop nesting section to `_index.md`). I could add the lack of error reporting to the loop nesting section as well to avoid repeating that in both `LOOP` and `LOOPI` documentation. I do think it's important the loop nesting section is linked somehow so anyone who just jumps to the `LOOP` and `LOOPI` documentation without reading all of `_index.md` realises there's more to read.

Reply:
Hmm, good point. Maybe something like "For more information about how to use LOOP / LOOP, see ..." ?